internal/factsengine/gatherers/ini_files_test.go

```diff
@@ -0,0 +1,256 @@
+	suite.NoErrorf(err, "error creating content01")
+
+	c := gatherers.NewIniFilesGatherer(fs)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "global conf",
+			Gatherer: "ini_files",
+			Argument: "global.ini",
+		},
+	}
+
+	factResults, err := c.Gather(context.Background(), factRequests)
+
+	suite.NoError(err)
+	suite.Len(factResults, 1)
+	suite.Empty(factResults[0].Error)
+	fact, ok := factResults[0].Value.(*entities.FactValueList)
+	if !ok {
+		suite.Fail("fact value is not a list")
+	}
+	suite.Len(fact.Value, 1)
+	suite.Equal(
+		fact.Value[0],
+		&entities.FactValueMap{
+			Value: map[string]entities.FactValue{
+				"sid": &entities.FactValueString{Value: "S01"},
+				"content": &entities.FactValueMap{
+					Value: map[string]entities.FactValue{
+						"key1": &entities.FactValueString{Value: "value1"},
+						"section1": &entities.FactValueMap{
+							Value: map[string]entities.FactValue{
+								"key2": &entities.FactValueString{Value: "value2"},
+							},
+						},
+						"section2": &entities.FactValueMap{
+							Value: map[string]entities.FactValue{
+								"key3": &entities.FactValueString{Value: "value3"},
+							},
+						},
+					},
+				},
+			},
+		})
+
+}
+
+func (suite *IniFilesTestSuite) TestIniFilesGathererGlobalIniMultiParse() {
+	content01 := "key1=value1"
+	content02 := "key2=value2"
+
+	fs := afero.NewMemMapFs()
+	err := afero.WriteFile(fs, "/usr/sap/S01/SYS/global/hdb/custom/config/global.ini", []byte(content01), 0400)
+	suite.NoErrorf(err, "error creating content01")
+	err = afero.WriteFile(fs, "/usr/sap/S02/SYS/global/hdb/custom/config/global.ini", []byte(content02), 0400)
+	suite.NoErrorf(err, "error creating content02")
+
+	c := gatherers.NewIniFilesGatherer(fs)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "global conf",
+			Gatherer: "ini_files",
+			Argument: "global.ini",
+		},
+	}
+
+	factResults, err := c.Gather(context.Background(), factRequests)
+
+	suite.NoError(err)
+	suite.Len(factResults, 1)
+	suite.Empty(factResults[0].Error)
+	suite.IsType(&entities.FactValueList{}, factResults[0].Value)
+	fact, ok := factResults[0].Value.(*entities.FactValueList)
+	if !ok {
+		suite.Fail("fact value is not a list")
+	}
+	suite.Len(fact.Value, 2)
+	suite.Equal(
+		fact.Value[0],
+		&entities.FactValueMap{
+			Value: map[string]entities.FactValue{
+				"sid": &entities.FactValueString{Value: "S01"},
+				"content": &entities.FactValueMap{
+					Value: map[string]entities.FactValue{
+						"key1": &entities.FactValueString{Value: "value1"},
+					},
+				},
+			},
+		})
+	suite.Equal(
+		fact.Value[1],
+		&entities.FactValueMap{
+			Value: map[string]entities.FactValue{
+				"sid": &entities.FactValueString{Value: "S02"},
+				"content": &entities.FactValueMap{
+					Value: map[string]entities.FactValue{
+						"key2": &entities.FactValueString{Value: "value2"},
+					},
+				},
+			},
+		})
+
+}
+
+func (suite *IniFilesTestSuite) TestIniFilesGathererGlobalIniPartial() {
```

Comment:
question: just curiosity. Why is this test sending a fact with an error?

Reply:
In fact, the error is `*entities.FactGatheringError)(nil)`. And it has a correct value.
I don't know if this test is really useful

Reply:
I changed the test because what I wanted to prove is how code handles the case in which one of the parsing fails